src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -75,7 +80,17 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+        Person editedPerson = null;
+        if (personToEdit instanceof Student) {
+            Student studentToEdit = (Student) personToEdit;
+            editedPerson = createEditedPerson(studentToEdit, editPersonDescriptor);
+        } else if (personToEdit instanceof Professor) {
+            Professor professorToEdit = (Professor) personToEdit;
+            editedPerson = createEditedPerson(professorToEdit, editPersonDescriptor);
+        } else if (personToEdit instanceof TeachingAssistant) {
+            TeachingAssistant taToEdit = (TeachingAssistant) personToEdit;
+            editedPerson = createEditedPerson(taToEdit, editPersonDescriptor);
+        }
```

Comment:
I dont think it is possible for this part as the argument needed for this function is `Model` which as no subclasses.

Reply:
Even if i dont put `instanceOf` in this function. I would probably need it in createdEditedPerson.

Reply:
But if you have an idea on how to use polymorphism please do tell me.

Reply:
This uses `type.equals()` which also not polymorphism. Isnt overriding better? I understand that this will make the `instanceOf() ` in `execute()` irrelevant and thus reduces the amount of code. So the issue is like if I don't put `instanceOf `in `execute()` it I would still need to put in `createEditedPerson()` hahaha. But if all you are ok with nitant suggestion I do not mind using his suggestion as it would indeed reduce the amount of code.